hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/map/AbstractMapQueryMessageTask.java

```diff
@@ -231,4 +246,26 @@ private void collectResultsFromMissingPartitions(BitSet finishedPartitions, Coll
             }
         }
     }
+
+    // for rolling-upgrade compatibility, can be deleted in 4.0
+    private Operation createQueryOperation(Query query, Version clusterVersion, MapServiceContext mapServiceContext) {
```

Comment:
This sounds legit, but then I'll have the same duplication in the MapOperationProvider and HDMapOperationProvider, so it's no good...
I'm planning to get rid of the duplication of the query logic in the AbstractMapQueryMessageTask, and use the MapQueryEngine there -> then I'll get rid of this duplication implicitly.